Client/Helpers/FxALoginHelper.swift

```diff
@@ -143,7 +143,7 @@ class FxALoginHelper {
         assert(profile != nil, "Profile should still exist and be loaded into this FxAPushLoginStateMachine")
 
         guard let profile = profile,
-            let account = FirefoxAccount.from(profile.accountConfiguration, andJSON: data) else {
+            let account = FirefoxAccount.from(profile.accountConfiguration, andJSON: data, getDeviceName: profile.syncManager.getClientName) else {
```

Comment:
`getDeviceName` is a pretty hacky hack.

Prior to device registration, there was only one name in this system: the name managed by Sync and stored in the Sync client record.

There are now two names: the Sync client name and the FxA device name.

Those are related, but perhaps not necessarily the same — this'll be "Fennec (rnewman) on Richard's iPhone" — and certainly it's weird that there's this bidirectional relationship: FxA passes the device ID into Sync, and here we pickle the Sync client name accessor to get the name out for FxA.

During FxA login you end up constructing the default Sync client name anyway.

It seems like in this work you're asserting that the FxA device name and the Sync client record entry should be the same. In that case, the right thing to do is to **move ownership of that value to FxA**, right next to the device registration, complete with the same force-upload logic, and thread it into Sync in exactly the same way: in the sync auth state.

That should make this PR much clearer and more understandable, no?